File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left_accum = 1
        right_accum = 1

        result = [1 for _ in range(len(nums))]
        
        for i in range(len(nums)):
            result[i] *= left_accum
            left_accum *= nums[i]

        for i in range(len(nums) - 1, -1, -1):
            result[i] *= right_accum
            right_accum *= nums[i]

        return result

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int nums_left = 1;
        int nums_right = 1;
        vector<int> result(nums.size(), 1);

        for (int i = 0; i < nums.size(); i++) {
            result[i] *= nums_left;
            nums_left *= nums[i];
        }

        for (int i = nums.size() - 1; i >= 0; i--) {
            result[i] *= nums_right;
            nums_right *= nums[i];
        }

        return result;
    }
};